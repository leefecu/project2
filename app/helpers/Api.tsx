import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
