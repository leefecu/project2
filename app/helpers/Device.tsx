import { NativeModules, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import CarrierInfo from 'react-native-carrier-info';
import RNFS from 'react-native-fs';

const NZ_MCC = '530';
const NZ_COUNTRY_CODE = 'nz';
const IS_IOS = Platform.OS === 'ios';

const NetworkInfo = NativeModules.RNNetworkInfo;

export const isSimulator = async (): Promise<boolean> => {
  const isEmulator = await DeviceInfo.isEmulator();
  return isEmulator;
};

export const isSimIn = async () => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return true;
  }

  try {
    // mobileCountryCode method fails when no SIM card in the device
    await CarrierInfo.mobileCountryCode();
    return true;
  } catch (e) {
    return false;
  }
};

export const checkForRoaming = async (debug: boolean = false): Promise<boolean> => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return debug;
  }

  try {
    if (Platform.OS === 'ios') {
      const operatorPlist = await NativeModules.NetworkInfo.getOperatorPlist();
      const mcc = operatorPlist.match(/\d+\/carrier.plist/g);
      if (mcc && mcc.length > 0) {
        return mcc[0].substr(0, 3) !== NZ_MCC;
      }
      return false;
    }
    const countryCode = await NetworkInfo.isoCountryCode();
    return countryCode !== NZ_COUNTRY_CODE;
  } catch (e) {
    return false;
  }
};

const majorVersionIOS: number = parseInt(`${Platform.Version}`, 10);

export const isAutoFillSupported: boolean = IS_IOS && majorVersionIOS >= 12;

// Figured out how current(old) DataClock app stores msisdn
// it's stored in xml file
// shared_prefs/nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml
const msisdnFileName = 'nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml';
const msisdnFolderName = 'shared_prefs';
const msisdnRegex = /com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">\d+<\/string/g;

export const getExistingMsisdn = async (): Promise<TMigrationData> => {
  if (IS_IOS) {
    try {
      const msisdn = await NativeModules.GetNumber.getMsisdn();
      return { number: msisdn, upgrade: !!msisdn };
    } catch (e) {
      return { number: '', upgrade: false };
    }
  }
  // This method only works with Android
  try {
    // e.g. RNFS.DocumentDirectoryPath = '/data/user/0/nz.co.twodegreesmobile.comptelfwd.smpl/files',
    const paths = RNFS.DocumentDirectoryPath.split('/');
    // need to remove "/files" and replace with "shared_prefs"
    paths[paths.length - 1] = msisdnFolderName;
    const path = paths.join('/');
    const xmlPath = `${path}/${msisdnFileName}`;
    const existingAccount = await RNFS.readFile(xmlPath);
    const match = msisdnRegex.exec(existingAccount);
    if (match && match.length > 0) {
      const msisdn = match[0]
        .replace('com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">', '')
        .replace('</string', '');

      return { number: msisdn, upgrade: true };
    }
    return { number: '', upgrade: false };
  } catch (e) {
    return { number: '', upgrade: false };
  }
};
