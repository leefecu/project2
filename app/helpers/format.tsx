export const getFormattedMobileNumber = (number: string): string => {
  const prefix = '+64';
  const numberParts = [];
  if (number.startsWith('+64')) {
    numberParts.push(number.substring(3, 5));
    numberParts.push(number.substring(5, 8));
    numberParts.push(number.substring(8, 13));
  } else if (number.startsWith('64')) {
    numberParts[0] = number.substring(2, 4);
    numberParts.push(number.substring(4, 7));
    numberParts.push(number.substring(7, 12));
  } else if (number.startsWith('0064')) {
    numberParts[0] = number.substring(4, 6);
    numberParts.push(number.substring(6, 9));
    numberParts.push(number.substring(9, 14));
  } else if (number.startsWith('02')) {
    numberParts[0] = number.substring(1, 3);
    numberParts.push(number.substring(3, 6));
    numberParts.push(number.substring(6, 12));
  } else if (number.startsWith('2')) {
    numberParts[0] = number.substring(0, 2);
    numberParts.push(number.substring(2, 5));
    numberParts.push(number.substring(5, 12));
  }

  return `${prefix} ${numberParts.join(' ')}`;
};

// number is always in the format of 22xxxyyyy
export const getLocalFormattedMobileNumber = (number: string): string => {
  return `0${number.substring(0, 2)} ${number.substring(2, 13)}`;
};

export const parseDuration = (duration: number): number =>
  duration % 60 === 0 ? duration / 60 : duration;

export const getUnit = (duration: number, short: boolean = false): string => {
  if (duration % 60 === 0) {
    const hour = duration / 60;
    return `${short ? 'hr' : 'hour'}${hour > 1 ? 's' : ''}`;
  }
  return `${short ? 'min' : 'minute'}${duration > 1 ? 's' : ''}`;
};

const DAY = 86400;
const HOUR = 3600;
const MIN = 60;

export const gerFormatedRemainingTime = (time: number): string => {
  let remainingTime = Math.ceil(Number(time));
  let displayFormatFactor = 0;
  const timeTokens = [];
  if (remainingTime >= DAY) {
    timeTokens.push(`${Math.floor(remainingTime / DAY)}d`);
    remainingTime %= DAY;
    displayFormatFactor = DAY;
  }
  if (remainingTime >= HOUR) {
    timeTokens.push(`${Math.floor(remainingTime / HOUR)}h`);
    remainingTime %= HOUR;
    if (!displayFormatFactor) {
      displayFormatFactor = HOUR;
    }
  }
  if (displayFormatFactor <= HOUR && timeTokens.length < 2 && remainingTime >= MIN) {
    timeTokens.push(`${Math.floor(remainingTime / MIN)}m`);
    remainingTime %= MIN;
    if (!displayFormatFactor) {
      displayFormatFactor = MIN;
    }
  }
  if (displayFormatFactor <= MIN && timeTokens.length < 2 && remainingTime > 0) {
    timeTokens.push(`${remainingTime}s`);
  }
  return timeTokens.join(' ');
};
