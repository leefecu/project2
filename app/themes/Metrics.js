import { Dimensions } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';

const { height, width } = Dimensions.get('window');

export default {
  buttonBorderRadius: 32,
  inputBorderRadius: 5,
  borderWidth: 1,
  smallestPadding: 4,
  smallerPadding: 8,
  smallPadding: 12,
  basePadding: 16,
  doublePadding: 24,
  largePadding: 36,
  largerPadding: 48,
  smallestMargin: 2,
  smallerMargin: 4,
  smallMargin: 8,
  baseMargin: 16,
  doubleMargin: 32,
  smallLineSpacing: 12,
  baseLineSpacing: 24,
  largeLineSpacing: 36,
  height,
  width,
  buttonMaxWidth: 340,
  smallButtonMaxWidth: 250,
  statusBarHeight: getStatusBarHeight(),
  smallIOSDevice: width <= 320,
  Fonts: {
    h1: 42,
    h2: 32,
    h3: 28,
    h4: 24,
    h5: 20,
    h6: 16,
    normal: 20,
    small: 17,
    smaller: 15,
    smallest: 13,
  },
};
