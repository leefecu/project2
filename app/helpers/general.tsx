import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
