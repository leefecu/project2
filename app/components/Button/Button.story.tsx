import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import Button from './Button';
import Metrics from '../../themes/Metrics';
import Styles from '../../Stories.styles';

const BUTTON_WIDTH = Metrics.width - Metrics.doublePadding * 2;
const SMALL_BUTTON_WIDTH = Metrics.width / 2;

storiesOf('Component | Button', module)
  .add('Primary', () => {
    return (
      <View style={Styles.container}>
        <View style={Styles.row}>
          <Text style={Styles.title}>Default</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={BUTTON_WIDTH}
          />
        </View>
        <View style={Styles.row}>
          <Text style={Styles.title}>Disabled</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={BUTTON_WIDTH}
            disabled
          />
        </View>
      </View>
    );
  })
  .add('Primary with Background', () => {
    return (
      <View style={Styles.containerWithBG}>
        <View style={Styles.row}>
          <Text style={Styles.title}>Default</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={BUTTON_WIDTH}
            withBG
          />
        </View>
        <View style={Styles.row}>
          <Text style={Styles.title}>Disabled</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={BUTTON_WIDTH}
            disabled
            withBG
          />
        </View>
      </View>
    );
  })
  .add('Small button', () => {
    return (
      <View style={Styles.containerWithBG}>
        <View style={Styles.row}>
          <Text style={Styles.title}>Default</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            smallButton
            width={SMALL_BUTTON_WIDTH}
          />
        </View>
        <View style={Styles.row}>
          <Text style={Styles.title}>Disabled</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={SMALL_BUTTON_WIDTH}
            smallButton
            disabled
          />
        </View>
      </View>
    );
  })
  .add('Small button with Background', () => {
    return (
      <View style={Styles.containerWithBG}>
        <View style={Styles.row}>
          <Text style={Styles.title}>Default</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            smallButton
            width={SMALL_BUTTON_WIDTH}
            withBG
          />
        </View>
        <View style={Styles.row}>
          <Text style={Styles.title}>Disabled</Text>
          <Button
            onPress={action('clicked-text')}
            id="Get Started"
            label="Get Started"
            width={SMALL_BUTTON_WIDTH}
            smallButton
            disabled
            withBG
          />
        </View>
      </View>
    );
  });
