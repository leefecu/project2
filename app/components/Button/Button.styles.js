import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.transparent,
    borderColor: Colors.primaryDark,
    borderRadius: Metrics.buttonBorderRadius,
    borderWidth: Metrics.borderWidth,
    paddingVertical: Metrics.smallPadding,
    alignItems: 'stretch',
  },
  containerWithBG: {
    borderColor: Colors.whiteBorder,
  },
  smallButton: {
    paddingVertical: Metrics.smallerPadding,
  },
  pressed: {
    borderColor: Colors.primaryDark,
  },
  disabled: {
    borderColor: Colors.primaryDarkDisabled,
  },
  disabledWithBG: {
    borderColor: Colors.whiteDisabled,
  },
  label: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.normal,
    fontWeight: 'bold',
    fontFamily: 'CircularPro-Bold',
    textAlign: 'center',
  },
  smallLabel: {
    fontSize: Metrics.Fonts.smaller,
  },
  labelWithBG: {
    color: Colors.white,
  },
  labelPressed: {
    color: Colors.white,
  },
  labelDisabled: {
    color: Colors.primaryDarkDisabled,
  },
  labelWithBGDisabled: {
    color: Colors.whiteBorder,
  },
});
