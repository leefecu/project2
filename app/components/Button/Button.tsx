import React, { useState } from 'react';
import { Text, TouchableHighlight } from 'react-native';

import { noop, testProps } from '../../helpers/general';

import Styles from './Button.styles';
import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

interface Props {
  disabled?: boolean;
  disregardDisablePress?: boolean;
  fixedWidth?: number;
  id: string;
  label: string;
  onPress: () => void;
  smallButton?: boolean;
  withBG?: boolean;
  width?: number;
}

const Button = ({
  disabled,
  disregardDisablePress,
  id,
  label,
  onPress,
  smallButton,
  withBG,
  width,
}: Props) => {
  const [buttonPressed, setPressed] = useState(false);
  const pressed = () => {
    setPressed(true);
  };
  const unPressed = () => {
    setPressed(false);
  };
  return (
    <TouchableHighlight
      activeOpacity={1}
      onPress={disabled && !disregardDisablePress ? noop : onPress}
      onShowUnderlay={pressed}
      onHideUnderlay={unPressed}
      underlayColor={disabled ? Colors.transparent : Colors.primaryDark}
      style={[
        { width },
        Styles.container,
        smallButton && Styles.smallButton,
        withBG && Styles.containerWithBG,
        buttonPressed && Styles.pressed,
        disabled && !withBG && Styles.disabled,
        disabled && withBG && Styles.disabledWithBG,
      ]}
      {...testProps(`${id} button`)}
    >
      <Text
        style={[
          Styles.label,
          smallButton && Styles.smallLabel,
          withBG && Styles.labelWithBG,
          buttonPressed && Styles.labelPressed,
          disabled && withBG && Styles.labelWithBGDisabled,
          disabled && !withBG && Styles.labelDisabled,
        ]}
      >
        {label}
      </Text>
    </TouchableHighlight>
  );
};

Button.defaultProps = {
  containerWidth: Metrics.width,
  disabled: false,
  disregardDisablePress: false,
  id: '',
  label: '',
  onPress: noop,
  smallButton: false,
  withBG: false,
  width: 255,
};

export default Button;
