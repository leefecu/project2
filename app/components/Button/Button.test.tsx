/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Button from './Button';

import { noop } from '../../helpers/general';

import Metrics from '../../themes/Metrics';

const BUTTON_WIDTH = Metrics.width - Metrics.doublePadding * 2;

describe('Button component', () => {
  test('render Primary active button', () => {
    const button = create(
      <Button onPress={noop} id="Get Started" label="Get Started" width={BUTTON_WIDTH} />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render Primary disabled button', () => {
    const button = create(
      <Button disabled onPress={noop} id="Get Started" label="Get Started" width={BUTTON_WIDTH} />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render Primary with BG button', () => {
    const button = create(
      <Button onPress={noop} id="Get Started" label="Get Started" withBG width={BUTTON_WIDTH} />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render Primary with BG disabled button', () => {
    const button = create(
      <Button
        disabled
        onPress={noop}
        id="Get Started"
        label="Get Started"
        withBG
        width={BUTTON_WIDTH}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render active small button', () => {
    const button = create(
      <Button
        onPress={noop}
        id="Get Started"
        label="Get Started"
        smallButton
        width={BUTTON_WIDTH}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render disabled small button', () => {
    const button = create(
      <Button
        disabled
        onPress={noop}
        id="Get Started"
        label="Get Started"
        smallButton
        width={BUTTON_WIDTH}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render active small button with BG', () => {
    const button = create(
      <Button
        onPress={noop}
        id="Get Started"
        label="Get Started"
        smallButton
        width={BUTTON_WIDTH}
        withBG
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render disabled small button with BG', () => {
    const button = create(
      <Button
        disabled
        onPress={noop}
        label="Get Started"
        smallButton
        withBG
        width={BUTTON_WIDTH}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('trigger onPress function upon press', () => {
    const onPressEvent = jest.fn();
    const component = shallow(
      <Button onPress={onPressEvent} id="Get Started" label="Get Started" width={BUTTON_WIDTH} />,
    );
    component.simulate('press');
    expect(onPressEvent).toHaveBeenCalled();
  });

  test('does not trigger onPress function upon press when disabled', () => {
    const onPressEvent = jest.fn();
    const component = shallow(
      <Button
        disabled
        onPress={onPressEvent}
        id="Get Started"
        label="Get Started"
        width={BUTTON_WIDTH}
      />,
    );
    component.simulate('press');
    expect(onPressEvent).not.toHaveBeenCalled();
  });
});
