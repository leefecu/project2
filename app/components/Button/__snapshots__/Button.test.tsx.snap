// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render Primary active button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      false,
      false,
      false,
      false,
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        false,
        false,
        false,
        false,
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render Primary disabled button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      false,
      false,
      false,
      Object {
        "borderColor": "rgba(0, 64, 113, 0.4)",
      },
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "rgba(0, 64, 113, 0.4)",
        },
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render Primary with BG button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      false,
      Object {
        "borderColor": "rgba(255, 255, 255, 0.7)",
      },
      false,
      false,
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        false,
        Object {
          "color": "#ffffff",
        },
        false,
        false,
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render Primary with BG disabled button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      false,
      Object {
        "borderColor": "rgba(255, 255, 255, 0.7)",
      },
      false,
      false,
      Object {
        "borderColor": "rgba(255, 255, 255, 0.4)",
      },
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        false,
        Object {
          "color": "#ffffff",
        },
        false,
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
        },
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render active small button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      Object {
        "paddingVertical": 8,
      },
      false,
      false,
      false,
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 15,
        },
        false,
        false,
        false,
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render active small button with BG 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      Object {
        "paddingVertical": 8,
      },
      Object {
        "borderColor": "rgba(255, 255, 255, 0.7)",
      },
      false,
      false,
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 15,
        },
        Object {
          "color": "#ffffff",
        },
        false,
        false,
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render disabled small button 1`] = `
<View
  accessibilityLabel="Get Started button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      Object {
        "paddingVertical": 8,
      },
      false,
      false,
      Object {
        "borderColor": "rgba(0, 64, 113, 0.4)",
      },
      false,
    ]
  }
  testID="Get Started button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 15,
        },
        false,
        false,
        false,
        Object {
          "color": "rgba(0, 64, 113, 0.4)",
        },
      ]
    }
  >
    Get Started
  </Text>
</View>
`;

exports[`Button component render disabled small button with BG 1`] = `
<View
  accessibilityLabel=" button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "width": 702,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "#004071",
        "borderRadius": 32,
        "borderWidth": 1,
        "paddingVertical": 12,
      },
      Object {
        "paddingVertical": 8,
      },
      Object {
        "borderColor": "rgba(255, 255, 255, 0.7)",
      },
      false,
      false,
      Object {
        "borderColor": "rgba(255, 255, 255, 0.4)",
      },
    ]
  }
  testID=" button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#004071",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 15,
        },
        Object {
          "color": "#ffffff",
        },
        false,
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
        },
        false,
      ]
    }
  >
    Get Started
  </Text>
</View>
`;
